components/net/indexeddb/idb_thread.rs

```diff
@@ -0,0 +1,343 @@
+        origin: ImmutableOrigin,
+        db_name: String,
+    ) -> Option<&mut IndexedDBEnvironment<HeedEngine>> {
+        let idb_description = IndexedDBDescription {
+            origin,
+            name: db_name,
+        };
+
+        self.databases.get_mut(&idb_description)
+    }
+
+    fn handle_sync_operation(&mut self, operation: SyncOperation) {
+        match operation {
+            SyncOperation::OpenDatabase(sender, origin, db_name, version) => {
+                let idb_description = IndexedDBDescription {
+                    origin,
+                    name: db_name,
+                };
+
+                let idb_base_dir = self.idb_base_dir.as_path();
+
+                match self.databases.entry(idb_description.clone()) {
+                    Entry::Vacant(e) => {
+                        let db = IndexedDBEnvironment::new(
+                            HeedEngine::new(idb_base_dir, &idb_description.as_path()),
+                            version.unwrap_or(0),
+                        );
+                        sender.send(db.version).unwrap();
+                        e.insert(db);
+                    },
+                    Entry::Occupied(db) => {
+                        sender.send(db.get().version).unwrap();
+                    },
+                }
+            },
+            SyncOperation::DeleteDatabase(sender, origin, db_name) => {
+                let idb_description = IndexedDBDescription {
+                    origin,
+                    name: db_name,
+                };
+                self.databases.remove(&idb_description);
+
+                // FIXME:(rasviitanen) Possible security issue?
+                let mut db_dir = self.idb_base_dir.clone();
+                db_dir.push(&idb_description.as_path());
+                if std::fs::remove_dir_all(&db_dir).is_err() {
+                    sender.send(Err(())).unwrap();
+                } else {
+                    sender.send(Ok(())).unwrap();
+                }
+            },
+            SyncOperation::HasKeyGenerator(sender, origin, db_name, store_name) => {
+                let store_name = SanitizedName::new(store_name);
+                let result = self
+                    .get_database(origin, db_name)
+                    .map(|db| db.has_key_generator(store_name))
+                    .expect("No Database");
+                sender.send(result).expect("Could not send generator info");
+            },
+            SyncOperation::Commit(sender, _origin, _db_name, _txn) => {
+                // FIXME:(arihant2math) This does nothing at the moment
+                sender
+                    .send(IndexedDBThreadReturnType::Commit(Err(())))
+                    .expect("Could not send commit status");
+            },
+            SyncOperation::UpgradeVersion(sender, origin, db_name, _txn, version) => {
+                self.get_database_mut(origin, db_name).map(|db| {
+                    db.version = version;
+                });
```

Comment:
```suggestion
                if let Some(db) = self.get_database_mut(origin, db_name) {
                    db.version = version;
                };
```

Reply:
And please make similar changes to other uses of map in this file that don't use the return value.